add a command to create a PVC frame

rkrenzler opened the pull request.

rkrenzler committed:
add frame creating command without GUI

rkrenzler committed:
add frame works. but looks ugly

rkrenzler committed:
a simple version of the "create frame" command  works

rkrenzler committed:
add comboboxes to the "add frame" command

rkrenzler committed:
change icon for "add frame" command

Also removed some unused code

rkrenzler committed:
add QtDesigner file form "add frame" dialog

rkrenzler committed:
add new icon to "add frame" command

rkrenzler committed:
Add LGPL note to the source files

skaiser commented on frame.py:
```diff
@@ -0,0 +1,225 @@
+
+class UnplausibleDimensions(Error):
+	"""Exception raised when dimensions are unplausible. For example if
+	outer diameter is larger than the iner one.
```
Nit: inner

skaiser commented on frame.py:
```diff
@@ -0,0 +1,225 @@
+			raise UnplausibleDimensions("The length LZ %smust be larger than 2*G %s"%(self.LZ, 2*self.G))
+	
+	def createPipes(self, group, convertToSolid):
+		# Calculate pipe lengthes
```
Nit: lengths

skaiser commented on frame.py:
```diff
@@ -0,0 +1,225 @@
+		ypipe.Placement = FreeCAD.Placement(FreeCAD.Vector(0, self.G,0), FreeCAD.Rotation(FreeCAD.Vector(1,0,0),-90), FreeCAD.Vector(0,0,0))
+		ypipe.Label = "y-"+ypipe.Label
+
+		# Add 3 clones for each x,y,z-type of axis. Place them on the edges of the quebe
```
quebe -> cube

skaiser commented on frame.py:
```diff
@@ -0,0 +1,225 @@
+		self.document = document
+		self.pipe_table = pipe_table
+		self.corner_table = corner_table
+		self.LX = tu("12 in")
```
Should there also be a 16 inch frame? That's the one I built because the [existing docs](http://opensourceecology.org/wiki/D3D_PVC_pipe_frame) mentioned using the same length of steel rod from the BOM (15.5 in)

skaiser commented on frame.py:
```diff
@@ -0,0 +1,225 @@
+import outerCorner
+import pipe as pipemod
+
+tu = FreeCAD.Units.parseQuantity
```
Why is this variable called tu? Would parse_quanity or pq be a better name?

skaiser commented on D3D_AddFrame.py:
```diff
@@ -48,42 +50,10 @@ def Activated(self):
-        #obj.addProperty("App::PropertyFile", "sourceFile", "importPart").sourceFile = D3DInit.__dir__ + '/Resources/cad/D3D_X_Axis_Simple.fcstd'
-        # TODO(kaisers): If first part
-        #obj.addProperty("App::PropertyBool","fixedPosition","importPart")
+	pipeTable = pipeGui.GuiCheckTable() # Open a CSV file, check its content, and return it as a CsvTable object.
```
Where is the CSV file located?

skaiser replied:
It looks like some imports like pipe, pipeGui outerCorner and outerCornerGui are be missing too.

rkrenzler replied:
They are located in the ose-piping-workbench

skaiser commented on frame.py:
```diff
@@ -0,0 +1,225 @@
+import outerCorner
+import pipe as pipemod
+
+tu = FreeCAD.Units.parseQuantity
```
Why is this variable called tu? Would parse_quanity or pq be a better name?

rkrenzler replied:
tu = To Units. But you are right. I will change it to a better name: _parseQuantity_.

skaiser commented on frame.py:
```diff
@@ -0,0 +1,225 @@
+
+class UnplausibleDimensions(Error):
+	"""Exception raised when dimensions are unplausible. For example if
+	outer diameter is larger than the iner one.
```
Nit: Typo "inner"

rkrenzler replied:
fixed

skaiser commented on frame.py:
```diff
@@ -0,0 +1,225 @@
+			raise UnplausibleDimensions("The length LZ %smust be larger than 2*G %s"%(self.LZ, 2*self.G))
+	
+	def createPipes(self, group, convertToSolid):
+		# Calculate pipe lengthes
```
Nit: Typo "lengths"

rkrenzler replied:
corrected

skaiser commented on frame.py:
```diff
@@ -0,0 +1,225 @@
+		ypipe.Placement = FreeCAD.Placement(FreeCAD.Vector(0, self.G,0), FreeCAD.Rotation(FreeCAD.Vector(1,0,0),-90), FreeCAD.Vector(0,0,0))
+		ypipe.Label = "y-"+ypipe.Label
+
+		# Add 3 clones for each x,y,z-type of axis. Place them on the edges of the quebe
```
Typo "quebe" -> "cube"

rkrenzler replied:
corrected

skaiser commented on frame.py:
```diff
@@ -0,0 +1,225 @@
+		self.document = document
+		self.pipe_table = pipe_table
+		self.corner_table = corner_table
+		self.LX = tu("12 in")
```
Should there also be a 16 inch frame? That's the one I built because the [existing docs](http://opensourceecology.org/wiki/D3D_PVC_pipe_frame) mentioned using the same length of steel rod from the BOM (15.5 in)

rkrenzler replied:
These values are only test values. There are used to easily optically recognize bugs in the code, if some dimensions will messed up or swapped.

skaiser commented on D3DBase.py:
```diff
@@ -0,0 +1,6 @@
+__dir__ = os.path.dirname(__file__)
+ICON_PATH = os.path.join( __dir__, 'Resources/icons' )
+IMAGE_PATH = os.path.join( __dir__, 'Resources/images' )
+TABLE_PATH = os.path.join( __dir__, 'tables' )
```
Should this go in Resources too?

rkrenzler replied:
You are right. But there is some difference between the table-resource and icons/image-resource.
The tables supposed to be editited but the other things not.

skaiser commented on frame.py:
```diff
@@ -0,0 +1,249 @@
+	document = FreeCAD.activeDocument()
+	pipe_table = CsvTable(PIPE_DIMENSIONS_USED)
+	corner_table = CsvTable(CORNER_DIMENSIONS_USED)
+	pipe_table.load(PIPE_CSV_TABLE_PATH)
```
I don't see where these two CSV table paths are defined. Am I missing something?

rkrenzler replied:
it is a bug. Thanks. I will fix it

skaiser commented on Resources/icons/AddFrame.svg:
```diff
@@ -0,0 +1,212 @@
+   id="svg6248"
+   sodipodi:version="0.32"
+   inkscape:version="0.92.2 (unknown)"
+   sodipodi:docname="CreateFrame.svg"
```
AddFrame.svg? Or it doesn't really matter here?

rkrenzler replied:
It does not influence the python code but I changed it for constancy reasons. (I make svg in Inkscape, but I did not know, that Inkscape keeps file name in such a way.)

skaiser reviewed (changes requested):
Nice work!!

One general comment: Should the file names reflect the actions are specific to PVC pipe frames? That is, addFrame -> addPVCframe or something?

My main question before approving the changes is about the location of the CSV files. They seem to be missing. Am I missing something?

skaiser commented:
It looks like some imports like pipe, pipeGui outerCorner and outerCornerGui are missing.

rkrenzler commented:
Hi Stephen,

the pipe, pipeGui, outerCorner, and outerCornerGui are parts of my
ose-piping-workbench.

Are there some smart way to resolve dependencies?

Ruslan


skaiser <notifications@github.com> schrieb am So., 25. Feb. 2018, 20:16:

> It looks like some imports like pipe, pipeGui outerCorner and
> outerCornerGui are missing.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/skaiser/D3D-Printer-Workbench/pull/1#issuecomment-368336143>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ATMY4oGYUCJE5uSB2Q5WkQoRA2fkXmA5ks5tYbGEgaJpZM4SR1bG>
> .
>

skaiser commented:
Ah, ok. I think the way to deal with that in Git is submodules:
https://github.com/blog/2104-working-with-submodules



On Sun, Feb 25, 2018 at 11:22 AM, rkrenzler <notifications@github.com>
wrote:

> Hi Stephen,
>
> the pipe, pipeGui, outerCorner, and outerCornerGui are parts of my
> ose-piping-workbench.
>
> Are there some smart way to resolve dependencies?
>
> Ruslan
>
>
> skaiser <notifications@github.com> schrieb am So., 25. Feb. 2018, 20:16:
>
> > It looks like some imports like pipe, pipeGui outerCorner and
> > outerCornerGui are missing.
> >
> > —
> > You are receiving this because you authored the thread.
> > Reply to this email directly, view it on GitHub
> > <https://github.com/skaiser/D3D-Printer-Workbench/pull/1#
> issuecomment-368336143>,
> > or mute the thread
> > <https://github.com/notifications/unsubscribe-auth/
> ATMY4oGYUCJE5uSB2Q5WkQoRA2fkXmA5ks5tYbGEgaJpZM4SR1bG>
> > .
> >
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/skaiser/D3D-Printer-Workbench/pull/1#issuecomment-368336561>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAlD2A3KcRvBmhB64nQUyDeDWZfXyjdVks5tYbLegaJpZM4SR1bG>
> .
>

rkrenzler commented:
skaiser <notifications@github.com> schrieb am So., 25. Feb. 2018, 20:55:

> Ah, ok. I think the way to deal with that in Git is submodules:
> https://github.com/blog/2104-working-with-submodules



Thanks for the link.

>
>
>
>
> On Sun, Feb 25, 2018 at 11:22 AM, rkrenzler <notifications@github.com>
> wrote:
>
> > Hi Stephen,
> >
> > the pipe, pipeGui, outerCorner, and outerCornerGui are parts of my
> > ose-piping-workbench.
> >
> > Are there some smart way to resolve dependencies?
> >
> > Ruslan
> >
> >
> > skaiser <notifications@github.com> schrieb am So., 25. Feb. 2018, 20:16:
> >
> > > It looks like some imports like pipe, pipeGui outerCorner and
> > > outerCornerGui are missing.
> > >
> > > —
> > > You are receiving this because you authored the thread.
> > > Reply to this email directly, view it on GitHub
> > > <https://github.com/skaiser/D3D-Printer-Workbench/pull/1#
> > issuecomment-368336143>,
> > > or mute the thread
> > > <https://github.com/notifications/unsubscribe-auth/
> > ATMY4oGYUCJE5uSB2Q5WkQoRA2fkXmA5ks5tYbGEgaJpZM4SR1bG>
> > > .
> > >
> >
> > —
> > You are receiving this because you commented.
> > Reply to this email directly, view it on GitHub
> > <
> https://github.com/skaiser/D3D-Printer-Workbench/pull/1#issuecomment-368336561
> >,
> > or mute the thread
> > <
> https://github.com/notifications/unsubscribe-auth/AAlD2A3KcRvBmhB64nQUyDeDWZfXyjdVks5tYbLegaJpZM4SR1bG
> >
> > .
> >
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/skaiser/D3D-Printer-Workbench/pull/1#issuecomment-368339177>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ATMY4m9b8VHBs5SGCVREmOqQ6bBwvvp6ks5tYbq-gaJpZM4SR1bG>
> .
>

rkrenzler committed:
change "tu" to a better name "parseQuantity"

rkrenzler committed:
fix typos and add comments.

Followed comments of @skaiser.

rkrenzler committed:
fix bug "bad CSV paths" in frame.TestTable

Followed comments of @skaiser.

rkrenzler committed:
change some parameters of AddFrame.svg.

This change does not influence the behaviour of the python code but adds
some consistancy. See comments of @skaiser.

rkrenzler commented:
You are right. I will change D3D_AddFrame to D3D_AddPvcFrame. (Not D3D_Add**PVC**Frame, because it is a common recommendation for abbreviations in a camel case notation, to treat abbreviation as words).

rkrenzler committed:
add Pvc prefixes

See comments from @skaiser.

rkrenzler commented:
I adjust the code taking in consideration your comment.

skaiser commented:
Thanks!

Can you do one more thing and update the README.md with instructions for
how to resolve the dependencies for your ose-piping-workbench? That way,
someone who tries to install this workbench knows how to get it working
since it won't be able to run without the ose-piping-workbench.

-Stephen




On Sat, Mar 3, 2018 at 4:35 AM, rkrenzler <notifications@github.com> wrote:

> I adjust the code taking in consideration your comment.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/skaiser/D3D-Printer-Workbench/pull/1#issuecomment-370144497>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAlD2EY659qX09Aw-Pdbyqoe7SVbKsRbks5tao4jgaJpZM4SR1bG>
> .
>

rkrenzler commented:
Hi Stephen,

2018-03-05 0:18 GMT+01:00 skaiser <notifications@github.com>:

> Thanks!
>
> Can you do one more thing and update the README.md with instructions for
> how to resolve the dependencies for your ose-piping-workbench? That way,
> someone who tries to install this workbench knows how to get it working
> since it won't be able to run without the ose-piping-workbench.
>
>
You are right. I also better need some kinde GUI message, which tells
"blabla bla no ose-piping-workbench😞. blablabla github, blablabla install
blablabla😄" to the user.



> -Stephen
>
Best regards,
Ruslan

>
>
>
>
> On Sat, Mar 3, 2018 at 4:35 AM, rkrenzler <notifications@github.com>
> wrote:
>
> > I adjust the code taking in consideration your comment.
> >
> > —
> > You are receiving this because you commented.
> > Reply to this email directly, view it on GitHub
> > <https://github.com/skaiser/D3D-Printer-Workbench/pull/1#
> issuecomment-370144497>,
> > or mute the thread
> > <https://github.com/notifications/unsubscribe-auth/AAlD2EY659qX09Aw-
> Pdbyqoe7SVbKsRbks5tao4jgaJpZM4SR1bG>
>
> > .
> >
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/skaiser/D3D-Printer-Workbench/pull/1#issuecomment-370273925>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ATMY4oNckPCHc3ZZJQfdo4hpIaUnTOhGks5tbHZjgaJpZM4SR1bG>
> .
>

rkrenzler committed:
add installation instructions for OSE-piping workbench

The OSE-piping workbench is required for creating of PVC frames.

rkrenzler commented:
I adjusted the README.md

rkrenzler commented:
Did I missed something?

rkrenzler committed:
use Jean-Babtiste's icon for creating of a frame

rkrenzler committed:
improve Add Frame dialog

add selection dialogs instead of combo list to
pipe and corner names

rkrenzler committed:
use smaller AddFrame icon from Jean-Baptiste
